pywr/recorders/_recorders.pyx

```diff
@@ -240,6 +250,78 @@ cdef class Recorder(Component):
             else:
                 raise ValueError("Objective direction type not recognised.")
 
+    property constraint_lower_bounds:
+        def __set__(self, value):
+            if value is None:
+                self._constraint_lower_bounds = -np.inf
+            else:
+                if self.constraint_upper_bounds is not None and value > self.constraint_upper_bounds:
+                    raise ValueError('Lower bounds can not be larger than the upper bounds.')
+                self._constraint_lower_bounds = value
+        def __get__(self):
+            if np.isneginf(self._constraint_lower_bounds):
+                return None
+            else:
+                return self._constraint_lower_bounds
+
+    property constraint_upper_bounds:
+        def __set__(self, value):
+            if value is None:
+                self._constraint_upper_bounds = -np.inf
+            else:
+                if self.constraint_lower_bounds is not None and value < self.constraint_lower_bounds:
+                    raise ValueError('Upper bounds can not be smaller than the lower bounds.')
+                self._constraint_upper_bounds = value
+        def __get__(self):
+            if np.isneginf(self._constraint_upper_bounds):
```

Comment:
should be `np.isinf`

Reply:
Nice catch. The internal representation should have been `np.inf` as well.